ping.go

```diff
@@ -339,6 +339,58 @@ func (p *Pinger) Addr() string {
 	return p.addr
 }
 
+// SetSrcAddr set source by interface name or ipv4 address
+// Interface name can be "eth0" or ip like "192.168.1.239"
+func (p *Pinger) SetSrcAddr(iface string) error {
+	if ip := net.ParseIP(iface); ip != nil {
+		if ip.To4() != nil {
+			p.SetNetwork("ipv4")
+		} else {
+			p.SetNetwork("ipv6")
+		}
+		p.Source = iface
+		return nil
+	}
+	if ipAddr, err := p.GetIfaceAddrs(iface); err != nil {
+		return err
+	} else {
+		p.Source = ipAddr[0].String() // As the private IP range is generally in ipv4, so as default it sets ipv4 of the iface
+		return nil
+	}
+}
+
+// GetIfaceAddrs is helper function which returns ipv4 and ipv6 as a array of net.IP type of supplied interface name
+// GetIfaceAddrs returns [ipv4,ipv6] of type net.IP which can be parsed to string
+func (p *Pinger) GetIfaceAddrs(interfaceName string) (addrs []net.IP, err error) {
```

Comment:
This is a helper function so let's make it private i.e. start with a lowercase letter.

Reply:
I think we can keep this as public, it can be used by the end user if needed to fetch IPv4 and IPv6 via interface.